ccm_web/package.json

```diff
@@ -18,6 +18,8 @@
     "css-loader": "^5.1.1",
     "express": "^4.17.1",
     "html-webpack-plugin": "^5.2.0",
+    "keycode-js": "^3.1.0",
+    "notistack": "^1.0.5",
```

Comment:
We should discuss the choice of these dependencies in a meeting and make sure they align with the rest of the project goals. `notistack` seems pretty neat, though it makes sense to do a little research to figure out what else is out there. I don't quite understand `keycode-js` (or the need for it) yet; seems like it's cool way to customize keyboard control, but customization may get in the way of accessibility (by violating expectations). I'm guessing (but could be wrong) that we can get by without it, and I generally would like to minimize dependencies.

Reply:
keycode-js is just a package that defines a bunch of constants so you can do this

```
  const keyPress = (code: string): void => {
    if (code === CODE_ENTER || code === CODE_NUMPAD_ENTER) {
      save()
    } else if (code === CODE_ESCAPE) {
      cancel()
    }
  }
```
instead of
```
  const keyPress = (code: string): void => {
    if (code === 'Enter' || code === 'NumpadEnter') {
      save()
    } else if (code === 'Escape') {
      cancel()
    }
  }
```

Reply:
I chose notistack becasue it was recommended by the [material-ui documentation](https://material-ui.com/components/snackbars/#notistack).  That carried enough weight for me but I didn't shop around, either.

Reply:
I do agree with @ssciolla that we don't want to pull a lot of dependencies since it make the main.js file bulk. `keycode-js` seems to be cool, but we can define our own constants as separate file and then just use what is needed.

Reply:
I think it's only [importing ](https://github.com/tl-its-umich-edu/canvas-course-manager-next/pull/47/files#diff-51ae223315fee7af3e83fc1b165b53f1857295b24e0dc9a5953253ebcc064cb1R6) what is used?  I'm not entirely sure how minimizing the amount of code imported works but I thought this was safe

Reply:
I guess I'd want to know why `keyPress` is even necessary (if it is, then something like this library might make sense). People who will be using keyboards to navigate the site would mostly be using tabs and Enter (probably not escape). Once they are done typing, I would think they would just hit Tab and then Enter once they're on the Save or Cancel buttons. Is that wrong or short-sighted somehow?

Reply:
It is so when the user is editing the course name escape will fire the cancel event (same as the cancel button) and Enter will complete the edit (same as the save button)

Reply:
Mmm, not sure. These behaviors seem reasonable to me, but UX team may say something else. I'm not familiar with regular uses of `esc` in an accessibility context. `Enter` is more common, and this logic seems right to me, but there may be another `onEnter` prop you could handle this with.

The concern with the dependency is that it's one more thing to keep updated/maintain, and you're only using it for two constants so far. If this were a game of some sort where you were using tons of keys, it would make mores sense to me. You could write your own constants here, or safely use the strings, right?

All that said, I don't see this is as a huge issue, or a blocker for me.